src/Routes/Devices/UpdateSystem.js

```diff
@@ -104,31 +134,105 @@ CurrentVersion.propTypes = {
   image: PropTypes.object,
 };
 
-const UpdateSystemMain = ({ data }) => {
+const UpdateSystemMain = ({ data, fetchDevices, isLoading, hasError }) => {
   const device = data?.Device;
+  const [selectedVersion, setSelectedVersion] = useState(null);
+  const [selectedCommitID, setSelectedCommitID] = useState(null);
+  const [isUpdateSubmitted, setIsUpdateSubmitted] = useState(false);
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const { pathname, search } = useLocation();
+  const match = useRouteMatch();
+
+  const setUpdateEvent = (value) => {
+    setSelectedVersion(value.cells[0]);
+    setSelectedCommitID(value);
+  };
+
+  const handleUpdateEvent = async () => {
+    setIsUpdateSubmitted(true);
+    const statusMessages = {
+      onInfo: {
+        title: 'Updating system',
+        description: ` ${selectedCommitID.deviceName} was added to the queue.`,
+      },
+      onError: {
+        title: 'Error',
+        description: `An error occurred making the request`,
```

Comment:
I would suggest adding more friendly customer message here
something like "Failed to update the selected device(s)"

Reply:
makes sense to me, i will handle it in a separate PR.